internal/osbuild2/dnf_config_stage.go

```diff
@@ -1,22 +1,49 @@
 package osbuild2
 
+import (
+	"fmt"
+)
+
 // DNFConfigStageOptions represents persistent DNF configuration.
 type DNFConfigStageOptions struct {
 	// List of DNF variables.
 	Variables []DNFVariable `json:"variables,omitempty"`
+	Config    *DNFConfig    `json:"config,omitempty"`
 }
 
 func (DNFConfigStageOptions) isStageOptions() {}
 
 // NewDNFConfigStageOptions creates a new DNFConfig Stage options object.
-func NewDNFConfigStageOptions(variables []DNFVariable) *DNFConfigStageOptions {
+func NewDNFConfigStageOptions(variables []DNFVariable, config *DNFConfig) *DNFConfigStageOptions {
 	return &DNFConfigStageOptions{
 		Variables: variables,
+		Config:    config,
+	}
+}
+
+func (o DNFConfigStageOptions) validate() error {
+	if o.Config != nil && o.Config.Main != nil {
+		valid := false
+		allowedIPR := []string{"4", "IPv4", "6", "IPv6"}
+		for _, v := range allowedIPR {
+			if o.Config.Main.IPResolve == v {
+				valid = true
```

Comment:
A very minor improvement, but we could `break` out of the loop as soon as it's found here.
We could also just return directly, but that might trip up potential future extensions to the function if we're not careful when adding them.